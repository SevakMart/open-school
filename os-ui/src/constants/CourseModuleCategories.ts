export const COURSE_CATEGORIES = [
  {
    value: '1',
    id: '1',
    title: 'Module 1',
  },
  {
    value: '2',
    id: '2',
    title: 'Module 2',
  },
  {
    value: '3',
    id: '3',
    title: 'Module 3',
  },
  {
    value: '4',
    id: '4',
    title: 'Module 4',
  },
];

export const COURSE_NAME = [
  {
    id: '1',
    desc: 'Additional Materials',
  },
  {
    id: '2',
    desc: 'Grade',
  },
  {
    id: '3',
    desc: 'Discussion Form',
  },
];
