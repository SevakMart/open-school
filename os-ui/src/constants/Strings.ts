// Navbar strings
export const APP_LOGO = 'Open-School';
export const EXPLORE_CATEGORIES = 'Explore Categories';
export const MENTORS = 'Mentors';
export const BECOME_A_MENTOR = 'Become a mentor';
export const SIGN_IN = 'Sign In';
export const ALL_LEARNING_PATHS = 'All Learning Paths';
export const MY_LEARNING_PATHS = 'My Learning Paths';
export const ALL = 'All';
export const IN_PROGRESS = 'InProgress';
export const COMPLETED = 'Completed';

// Header strings
export const FREE_EDUCATIONAL_PLATFORM = 'Educational Platform';
export const SIGN_UP = 'SIGN UP';
export const HEADER_INTRODUCTION = 'Choose Categories You Are Interested In Choose Categories You Are Interested InChoose Categories You Are Interested';
export const EDUCATION_PLATFORM_IMAGE = 'https://image.shutterstock.com/image-vector/ebook-logo-design-vector-electronic-260nw-1767373559.jpg';
export const CHOOSE_CATEGORIES_HEADER = 'Choose Categories You Are Interested In';

// URLS
export const GET_MENTORS_URL = '../../mentorData.json';
export const GET_CATEGORIES_URL = '../../categorieData.json';
export const GET_REAL_MENTORS_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users/mentors?`;
export const REGISTRATION_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/auth/register`;
export const GET_MAIN_CATEGORIES_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/categories?`;
export const GET_CATEGORY_SUBCATEGORY_SEARCH_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/categories/subcategories?title=`;
export const SIGNIN_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/auth/login`;
export const SAVE_PREFERRED_CATEGORIES = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users`;
export const USER_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users`;
// Error Message
export const ERROR_MESSAGE = 'Something went wrong please refresh the page :(';
export const EMPTY_DATA_ERROR_MESSAGE = 'No data to display :(';

// Successful Sign in message
export const SUCCESSFUL_SIGNIN_MESSAGE = 'You have Successfully signed in!';

// Regex for form validation
export const fullNameRegex = /^[A-Z\s]*$/i;
export const emailRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
export const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
/* Userâ€™s Password should have at least 8 characters,
including upper and lower case letters, numbers and signs (required) */

// Form Validation error message
export const INVALID_FULL_NAME_ERROR_MESSAGE = 'Invalid name or surname';
export const INVALID_EMAIL_ERROR_MESSAGE = 'Invalid email address';
export const INVALID_PASSWORD_ERROR_MESSAGE = 'Password should be at least 8 characters and contain at least one uppercase,lowercase,number and special character';
export const FULL_NAME_TOO_SHORT = 'full name is too short';
export const EMAIL_TOO_SHORT = 'email is too short';
export const PASSWORD_TOO_SHORT = 'password is too short';
export const PASSWORD_REQUIRED = 'password is required';
export const EMAIL_REQUIRED = 'email is required';

// Other non-specific strings
export const NO_COURSES_YET = 'No Courses Yet';
export const EXPLORE_COURSES = 'Explore Courses';
export const SUGGESTED_LEARNING_PATHS = 'Suggested Learning Paths';
export const OVERALL_PROGRESS = 'Overall Progress';
export const REMAINING_TIME = 'Remaining Time';
export const STATUS = 'Status';
export const DUE_DATE = 'Due Date';
export const RESUME_COURSE = 'Resume Course';
export const GRADE = 'Grade';
export const RATE_COURSE = 'Rate Course';
