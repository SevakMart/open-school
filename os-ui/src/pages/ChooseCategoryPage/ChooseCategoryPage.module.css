.buttonContainer {
  margin: 24px 88px;
  box-sizing: border-box;
  width: 108px;
  float: right;
}/*# sourceMappingURL=ChooseCategoryPage.module.css.map */