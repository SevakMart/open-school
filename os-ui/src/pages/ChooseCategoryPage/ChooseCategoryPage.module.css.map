{"version":3,"sources":["ChooseCategoryPage.module.scss","ChooseCategoryPage.module.css"],"names":[],"mappings":"AAIA;EACI,iBAAA;EACA,sBAAA;EACA,YAAA;EACA,YAAA;ACHJ","file":"ChooseCategoryPage.module.css"}