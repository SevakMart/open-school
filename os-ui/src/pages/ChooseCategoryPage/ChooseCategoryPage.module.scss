@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";

.mainHeader{
    @include fontStyle('Montserrat',normal,800,3rem,3.688rem);
    color:black;
    padding-left:6%;
    width:45%;
    margin-top:4%;
    margin-bottom:4%;
}
.categoriesList{
    @include flexStyle(row,none,space-between);
    flex-wrap:wrap;
    padding-left: 6%;
    padding-right:6%;
    margin-top:5%;
}
.nextButton{
    @include fontStyle('Montserrat',normal,500,1rem,1.25rem);
    @include buttonStyle(#5E617B,#FFFFFF,1rem,6.75rem,4rem,2rem,2rem,1.375rem,1.375rem);
    margin-left:88%;
    margin-bottom:6%;
}