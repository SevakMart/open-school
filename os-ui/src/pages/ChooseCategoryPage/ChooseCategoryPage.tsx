import { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import Header from './Subcomponents/Header/Header';
import MainContent from './Subcomponents/MainContent/MainContent';
import Button from '../../component/Button/Button';
import categoriesService from '../../services/categoriesService';
import { SearchedCategoryType } from '../../types/SearchedCategoryType';
import styles from './ChooseCategoryPage.module.scss';
import NavbarOnSignIn from '../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';

/* eslint-disable max-len */

const ChooseCategoryPage = ({ userInfo }:{userInfo:object}) => {
  const { t } = useTranslation();
  const [title, setTitle] = useState('');
  const [errorMessage, setErrorMessage] = useState('');
  const [searchedCategories, setSearchedCategories] = useState<SearchedCategoryType>({});
  const [isLoading, setIsLoading] = useState(true);
  const { buttonContainer } = styles;
  const handleChangeUrlTitleParam = (titleParam:string) => {
    setTitle(titleParam);
  };

  useEffect(() => {
    let cancel = false;
    categoriesService.getSearchedCategories({ title }, (userInfo as any).token)
      .then((data) => {
        if (cancel) return;
        if (!Object.entries(data).length) {
          setErrorMessage(t('error'));
          setIsLoading(false);
        } else if (!data.errorMessage) {
          setSearchedCategories({ ...data });
          setIsLoading(false);
          setErrorMessage('');
        } else {
          setErrorMessage(data.errorMessage);
          setIsLoading(false);
        }
      });
    return () => { cancel = true; };
  }, [title]);
  return (
    <>
      <NavbarOnSignIn />
      <Header changeUrlQueries={handleChangeUrlTitleParam} />
      <MainContent
        isLoading={isLoading}
        errorMessage={errorMessage}
        searchedCategories={searchedCategories}
      />
      <div className={buttonContainer}>
        <Button.NextButton userInfoId={(userInfo as any).id} userInfoToken={(userInfo as any).token} className={['nextButton']}>
          {t('button.chooseCategories.next')}
        </Button.NextButton>
      </div>
    </>
  );
};
export default ChooseCategoryPage;
