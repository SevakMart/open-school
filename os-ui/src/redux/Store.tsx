import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
