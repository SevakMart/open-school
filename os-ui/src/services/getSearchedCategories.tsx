import { fetchData } from './fetchData';

export const getSearchedCategories = async (url:string) => {
  const data = await fetchData(url);
  return data;
};
