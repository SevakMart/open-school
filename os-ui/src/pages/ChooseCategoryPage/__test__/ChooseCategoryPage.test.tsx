import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import ChooseCategoryPage from '../ChooseCategoryPage';
import { EMPTY_DATA_ERROR_MESSAGE } from '../../../constants/Strings';
import { store } from '../../../redux/Store';
import * as fetchSubcategories from '../../../services/getSearchedCategories';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
}));
const data = {
  Java: [
    {
      id: 1,
      title: 'Arrays',
    },
    {
      id: 2,
      title: 'Functions',
    },
    {
      id: 3,
      title: 'Classes',
    },

  ],
  Python: [
    {
      id: 4,
      title: 'Python Arrays',
    },
    {
      id: 5,
      title: 'Python Functions',
    },
    {
      id: 6,
      title: 'Python Classes',
    },
  ],
};

describe('Create test case to ChooSecategoryPage', () => {
  test('Create a snapshot test', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ChooseCategoryPage />
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
  });
  test('Check the content of the component', async () => {
    jest.spyOn(fetchSubcategories, 'getSearchedCategories').mockResolvedValue(data);
    expect.hasAssertions();
    render(
      <Provider store={store}>
        <ChooseCategoryPage />
      </Provider>,
    );
    const headerElement1 = await screen.findByTestId('Java');
    const headerElement2 = await screen.findByTestId('Python');
    const labelElement1 = await screen.findByTestId('Functions');
    const labelElement2 = await screen.findByTestId('Python Functions');

    expect(headerElement1).toBeInTheDocument();
    expect(headerElement1).toHaveTextContent('Java');
    expect(labelElement1).toBeInTheDocument();
    expect(labelElement1).toHaveTextContent('Functions');
    expect(headerElement2).toBeInTheDocument();
    expect(headerElement2).toHaveTextContent('Python');
    expect(labelElement2).toBeInTheDocument();
    expect(labelElement2).toHaveTextContent('Python Functions');
  });
  test('Test the output of page when the searched data is not found', async () => {
    jest.spyOn(fetchSubcategories, 'getSearchedCategories').mockResolvedValue({});
    expect.hasAssertions();
    render(
      <Provider store={store}>
        <ChooseCategoryPage />
      </Provider>,
    );
    const errorMessageHeader = await screen.findByTestId('chooseSubcategoriesErrorMessage');
    expect(errorMessageHeader).toBeInTheDocument();
    expect(errorMessageHeader).toHaveTextContent(EMPTY_DATA_ERROR_MESSAGE);
  });
});
