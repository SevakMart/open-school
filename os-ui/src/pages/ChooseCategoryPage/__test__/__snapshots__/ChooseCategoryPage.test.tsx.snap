// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test case to ChooSecategoryPage Create a snapshot test 1`] = `[Function]`;
